File: src/EnSerioSalesPredictor.Api/Dtos/CreateOrderDto.cs
namespace EnSerioSalesPredictor.Api.Dtos;

public class CreateOrderDto
{
    public int CustomerId { get; set; }
    public int EmpId { get; set; }
    public int ShipperId { get; set; }
    public required string ShipName { get; set; }
    public required string ShipAddress { get; set; }
    public required string ShipCity { get; set; }
    public DateTime OrderDate { get; set; }
    public required string RequiredDate { get; set; }
    public required string ShippedDate { get; set; }
    public decimal Freight { get; set; }
    public required string ShipCountry { get; set; }
    public required CreateOrderDetailsDto createOrderDetails { get; set; }
}

File: src/EnSerioSalesPredictor.Client/src/app/Models/CreateOrderDto.ts
import { CreateOrderDetailsDto } from "./CreateOrderDetailsDto";

export interface CreateOrderDto {
  customerId: number;
  empId: number;
  shipperId: number;
  shipName: string;
  shipAddress: string;
  shipCity: string;
  orderDate: Date;
  requiredDate: string;
  shippedDate: string;
  freight: number;
  shipCountry: string;
  createOrderDetails: CreateOrderDetailsDto;
}